kolibri/plugins/document_pdf_render/assets/src/module.js

```diff
@@ -1,16 +1,15 @@
 
 const KolibriModule = require('kolibri_module');
+const PDFComponent = require('./vue/index');
+const Vue = require('vue');
 
 class DocumentPDFModule extends KolibriModule {
-
-  render(contentData, renderArea) {
-    const documentTag = document.createElement('a');
-    documentTag.href = 'http://www.benlandis.com/sheet-music/piano-rickroll.pdf';
-    documentTag.textContent = contentData.id;
-    const titleTag = document.createElement('h1');
-    titleTag.innerHTML = contentData.id;
-    renderArea.appendChild(titleTag);
-    renderArea.appendChild(documentTag);
+  render(containerElement) {
+    const options = {
+      el: containerElement,
```

Comment:
I believe a more vuey way would be to define a core helper method which returns an appropriate renderer vue component based on the content type. Then `content-renderer` would be responsible for attaching it to the right element on its own, preventing internal DOM info from "leaking".

E.g.:

```html
<template>

  <div>
    <h3>
      {{ title }}
    </h3>
    <div v-el:container></div>
  </div>

</template>


<script>

  const Kolibri = require('kolibri');

  module.exports = {
    props: [
      'channelId',
      'contentId',
      'kind',
      'extension',
      'title',
      'contentData',
    ],
    ready() {
      Kolibri.getRendererComponentAsync(this.kind, this.extension, (Renderer) => {
        if (Renderer) {
          new Renderer({
            el: this.$els.container,
            propsData: {
              contentData: this.contentData,
            },
          });
        } else {
          // no renderer available
        }
      });
    },
  };

</script>


<style lang="stylus" scoped></style>
```